commands/command_filter.go

```diff
@@ -0,0 +1,207 @@
+		// _, err := io.Copy(os.Stdout, mr)
+		// if err != nil {
+		// Panic(err, "Error writing data to stdout:")
+		// }
+		return []byte("TODO"), nil
+	}
+
+	lfs.LinkOrCopyFromReference(ptr.Oid, ptr.Size)
+
+	cb, file, err := lfs.CopyCallbackFile("smudge", filename, 1, 1)
+	if err != nil {
+		Error(err.Error())
+	}
+
+	cfg := config.Config
+	download := lfs.FilenamePassesIncludeExcludeFilter(filename, cfg.FetchIncludePaths(), cfg.FetchExcludePaths())
+
+	if filterSmudgeSkip || cfg.GetenvBool("GIT_LFS_SKIP_SMUDGE", false) {
+		download = false
+	}
+
+	buf := new(bytes.Buffer)
+	err = ptr.Smudge(buf, filename, download, cb)
+	if file != nil {
+		file.Close()
+	}
+
+	if err != nil {
+		// Download declined error is ok to skip if we weren't requesting download
+		if !(errutil.IsDownloadDeclinedError(err) && !download) {
+			LoggedError(err, "Error downloading object: %s (%s)", filename, ptr.Oid)
+			if !cfg.SkipDownloadErrors() {
+				// TODO: What to do best here?
+				os.Exit(2)
+			}
+		}
+
+		return []byte(ptr.Encoded()), nil
+	}
+
+	return buf.Bytes(), nil
+}
+
+func filterCommand(cmd *cobra.Command, args []string) {
+	requireStdin("This command should be run by the Git filter")
+	lfs.InstallHooks(false)
+
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		buf := make([]byte, 4)
+		readBytes, err := reader.Read(buf)
+		if readBytes == 0 {
+			return
+		}
+		command := binary.LittleEndian.Uint32(buf)
+
+		if err != nil {
+			continue
+		}
+
+		// Read fileName length
+		buf = make([]byte, 4)
+		_, err = reader.Read(buf)
+		if err != nil {
+			Panic(err, "Error reading inputData filename length for cleaning.")
+		}
+		fileNameLen := binary.LittleEndian.Uint32(buf)
+
+		// Read fileName
+		fileName := ""
+		if fileNameLen > 0 {
+			buf = make([]byte, fileNameLen)
+			_, err = reader.Read(buf)
+			if err != nil {
+				Panic(err, "Error reading inputData filename for cleaning.")
+			}
+			fileName = string(buf)
+		}
+
+		// Read inputData length
+		buf = make([]byte, 4)
+		_, err = reader.Read(buf)
+		if err != nil {
+			Panic(err, "Error reading inputData pointer length for smudging.")
+		}
+		inputDataPtrLen := binary.LittleEndian.Uint32(buf)
+
+		// Read inputData
+		var outputData []byte
+		if inputDataPtrLen > 0 {
+			buf = make([]byte, inputDataPtrLen)
+			_, err = reader.Read(buf)
+			if err != nil {
+				Panic(err, "Error reading inputData pointer for smudging.")
+			}
+			inputData := bytes.NewReader(buf)
+
+			switch command {
+			case 1:
+				outputData, _ = clean(inputData, fileName)
+			case 2:
+				outputData, _ = smudge(inputData, fileName)
+			default:
+				panic("Unrecognized filter command")
+			}
+		}
+
+		writer := bufio.NewWriter(os.Stdout)
+		resLength := uint32(len(outputData))
+		binary.Write(writer, binary.LittleEndian, resLength)
+		if resLength > 0 {
+			writer.Write(outputData)
+		}
+		writer.Flush()
+	}
+}
+
+func init() {
+	filterCmd.Flags().BoolVarP(&filterSmudgeSkip, "skip", "s", false, "")
```

Comment:
Ah, I just got what you mean. I will rename the parameter to `--skip-smudge`, ok?